CMakeLists.txt

```diff
@@ -849,6 +856,70 @@ ELSEIF ("${CMAKE_SYSTEM_NAME}" MATCHES "SunOS")
     ENDIF (LIBUV_FOUND)
 ENDIF ()
 
+# h2olog
+# If BCC is installed to a non-default directory,
+# set something like `PKG_CONFIG_PATH=/path/to/bcc/lib/pkgconfig cmake ...`
+IF (CMAKE_SYSTEM_NAME STREQUAL "Linux" AND PKG_CONFIG_FOUND)
+    PKG_CHECK_MODULES(LIBBCC libbcc)
+
+    EXECUTE_PROCESS(
+        COMMAND uname -r
+        OUTPUT_VARIABLE KERNEL_VERSION
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    SET(REQUIRED_KERNEL_VERSION "4.9.0")
+    IF (KERNEL_VERSION VERSION_GREATER REQUIRED_KERNEL_VERSION OR KERNEL_VERSION VERSION_EQUAL REQUIRED_KERNEL_VERSION)
+        SET(HAVE_BPF_SUPPORTED_KERNEL ON)
+        MESSAGE(STATUS "Checking Linux kernel version ${KERNEL_VERSION} >= ${REQUIRED_KERNEL_VERSION} for h2olog - OK")
+    ENDIF ()
+
+    IF (WITH_DTRACE AND LIBBCC_FOUND AND HAVE_BPF_SUPPORTED_KERNEL)
+        SET(WITH_H2OLOG_DEFAULT ON)
```

Comment:
I might suggest checking if support for C++11 is available here. You can do something like:
```
IF (NOT (CMAKE_VERSION VERSION_LESS "3.8")) 
    CMAKE_POLICY(SET CMP0057 NEW)
    IF ("cxx_std_11" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
        MESSAGE(STATUS "C++11 OK")
    ENDIF ()
ENDIF ()
```
As h2olog is not a mandatory program, we can just disable it by default when CMake is old.

Reply:
It makes sense.

6e668c9bb0196e0f5953539b9083bd84e90d3f9b

Also, clarified a  required cmake version in h2olog.mt: 87fd4404487f864e95fa604cfd4d3a1926feedca